src/main/kotlin/org/jetbrains/research/testspark/tools/kex/KexSettingsArguments.kt

```diff
@@ -0,0 +1,111 @@
+package org.jetbrains.research.testspark.tools.kex
+
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.roots.CompilerModuleExtension
+import com.intellij.openapi.roots.ModuleRootManager
+import com.intellij.openapi.vfs.VfsUtil
+import org.jetbrains.research.testspark.bundles.kex.KexDefaultsBundle
+import org.jetbrains.research.testspark.settings.kex.KexSettingsState
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class KexSettingsArguments(
+    private val javaExecPath: String,
+    private val module: Module,
+    private val target: String,
+    private val resultName: String,
+    private val kexSettingsState: KexSettingsState,
+    private val kexExecPath: String,
+    private val kexHome: String,
+) {
+    fun buildCommand(): MutableList<String> {
+        return mutableListOf<String>(javaExecPath)
+            .setJvmProperties()
+            .setAddOpensArgs()
+            .setKexParams()
+            .setKexIniOptions()
+    }
+
+    private fun MutableList<String>.setJvmProperties(): MutableList<String> {
+        return this.addAll(
+            "-Xmx${KexDefaultsBundle.get("heapSize")}g",
+            "-Djava.security.manager",
+            "-Djava.security.policy==$kexHome/kex.policy",
+            "-Dlogback.statusListenerClass=ch.qos.logback.core.status.NopStatusListener",
+        )
+    }
+
+    private fun MutableList<String>.setAddOpensArgs(): MutableList<String> {
+        File("$kexHome/runtime-deps/modules.info").readLines().forEach { this.addAll("--add-opens", it) }
```

Comment:
Will this work if project uses JDK 8?

Reply:
Is there some special requirement for JDK 8?

Reply:
JVM module system was added only in JDK 9, so there is no `--add-opens` option for JDK 8. So we need to check if JDK 8 just ignores those options or fails with an error

Reply:
`kex.py` has a jvm version check allowing version 8 or above. I do the same. Should I make it strictly above 8?

Reply:
Yes, `kex.py` allows only versions 8 and above. However, it only adds `--add-opens` flags when the version is strictly greater than 8. So, we either have to replicate that behaviour, or we should ensure that it is not necessary